Structure_Engine/Query/SelectCase.cs

```diff
@@ -40,11 +41,19 @@ public static partial class Query
         /***************************************************/
 
         [Description("Filters out results from a specific case given an identifier, Loadcase or LoadCombination. For Loadcase and LoadCombination the number property will be used as identifier for filtering.")]
-        [Input("results","The list of results to filter.")]
+        [Input("results", "The list of results to filter.")]
         [Input("loadcase", "The case or combination to filter by. Should either be a string/int as identifier of the case or a Loadcase/LoadCombination where the number will be used as the identifier. If identifier can be extracted, all results are returned.")]
-        [Output("results","The filtered results. If no filtering param could be extracted, all results are returned.")]
+        [Output("results", "The filtered results. If no filtering param could be extracted, all results are returned.")]
         public static List<T> SelectCase<T>(this List<T> results, object loadcase) where T : IResult
         {
+            if (results.IsNullOrEmpty())
+                return null;
+            else if (loadcase == null)
+            {
+                Reflection.Compute.RecordError("Loadcase provided is null, therefore SelectCase cannot be evaluated.");
+                return null;
+            }
+
```

Comment:
This is handled below. Providing no case filters returns everything, which was the intended behaviour of this method.
```suggestion
```